docs/user_guide/architecture.rst

```diff
@@ -0,0 +1,128 @@
+      during project creation, these parameters are set in one of these
+      configuration files and can be passed to the simulation environment;
+      they might also contain parameters that aid in building various block designs
+
+   -  ``tests`` --- main simulation files, they control and monitor the DUT(s)
+
+   -  ``waves`` --- waveforms for various configuration options (only used in
+      GUI mode)
+
+   -  ``Makefile`` --- lists the required System Verilog files, Tcl scripts,
+      libraries used, and configuration-test_program combinations (if it’s the
+      case, otherwise all configurations run one and the same default test program)
+
+   -  ``system_project.tcl`` --- creates the project, the base block design based
+      on the test_harness_system_bd and sets up a couple of definitions for the
+      environment and test program
+
+   -  ``system_bd.tcl`` --- adds the DUT(s) to the block design with the configuration
+      files (even if no configuration parameter is used, a configuration file must
+      exist)
+
+   -  ``system_tb.sv`` --- connects the block design with the test program
+
+-  ``common`` --- common files that are used in many projects
+
+    -   ``sv`` --- basic functions, abstract/base classes, register maps to access
+        IP’s registers, classes for controlling the IPs
+
+    -   ``test_harness`` --- base designs
+
+        -   ``jesd_exerciser.tcl`` --- used to instantiate the ADI JESD204 Framework 
+        -   ``test_harness_system_bd.tcl`` --- used in all projects as a base design
+
+-  ``scripts`` --- used for creating and running the testbenches
+
+Creating a new testbench
+-------------------------------------------------------------------------------
+
+-  Create the ``Makefile`` and list all of the currently known dependencies
+-  Create the ``system_project.tcl`` script and add the currently known
+   simulation dependencies there as well
+-  Create at least one ``configuration file`` (it can be left empty)
+-  Create the ``system_bd.tcl`` script, which creates the block design
+-  Create the ``system_tb.sv`` file that connects the testbench with the block design
+-  Consider creating ``classes`` that encapsulate the variables and functions, so it
+   can be used in other testbenches
+-  Create the ``test_program.sv`` to run the simulation
+-  Update the ``Makefile`` and ``system_project.tcl`` with the new dependencies
+   if needed.
+
+Adding a new configuration to a test bench
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+   Simple, all of the configurations are compatible with all of the test programs.
+
+Requirements:
+
+-  New ``configuration`` doesn't change the connections between the block design and
+   the ``test program``.
+-  ``Test program`` can be used with the new ``configuration`` without modifications.
+-  Create a ``new configuration`` in ``cfg`` folder.
```

Comment:
Add: - If new ``parameter`` needs to be added, make sure all of the other configurations are updated with the new parameter name and a new value as well
Process: